manticore/seth.py

```diff
@@ -885,188 +977,253 @@ def global_findings(self):
-        #Try to recover metadata from solidity based contracts
-        try:
-            md_name, md_source_code, md_init_bytecode, md_metadata, md_metadata_runtime, md_hashes = self.context['seth']['metadata'][last_address]
-        except:
-            md_name, md_source_code, md_init_bytecode, md_metadata, md_metadata_runtime, md_hashes = None, None, None, None, None, None 
-
-        # try to get the runtime bytecode from the account
-        try:
-            runtime_bytecode = world.storage[last_address]['code']
-        except:
-            runtime_bytecode = ''   
-
-        e = state.context['last_exception']
-        if ty is not None:
-            if str(e) != ty:
-                return
-
-        output.write("REPORT:" + str(e)) 
-
-        try:
-            # Magic number comes from here:
-            # http://solidity.readthedocs.io/en/develop/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode
-            asm = list(evm.EVMAsm.disassemble_all(runtime_bytecode[:-9-33-2]))
-            asm_offset = 0
-            pos = 0
-            source_pos = md_metadata_runtime[pos]
-            for i in asm:
-                if len(md_metadata_runtime[pos]):
-                    source_pos = md_metadata_runtime[pos]
-                if asm_offset == last_pc:
-                    break
-                asm_offset += i.size
-                pos +=1
-
-            beg, size = map(int, source_pos.split(':'))
-
-            output.write( " at:" )
-            nl = md_source_code.count('\n')
-            snippet = md_source_code[beg:beg+size]
-            for l in snippet.split('\n'):
-                output.write('    %s  %s\n'%(nl, l))
-                nl+=1
-        except Exception as e:
-            output.write('\n')
-
-        output.write("BALANCES\n")
-        for address, account in world.storage.items():
-            if isinstance(account['balance'], Constant):
-                account['balance'] = account['balance'].value
-
-            if issymbolic(account['balance']):
-                m, M = solver.minmax(world.constraints, arithmetic_simplifier(account['balance']))
-                if m == M:
-                    output.write('\t%x %r\n'%(address, M))
-                else:
-                    output.write('\t%x range:[%x, %x]\n'%(address, m, M))
-            else:
-                output.write('\t%x %d wei\n'%(address, account['balance']))
-
-        if world.logs:
-            output.write('LOGS:\n')
-            for address, memlog, topics in world.logs:
-                try:
-                    res = memlog
-                    if isinstance(memlog, Expression):
-                        res = state.solve_one(memlog)
-                        if isinstance(memlog, Array):
-                            state.constrain(compare_buffers(memlog, res))
-                        else:
-                            state.constrain(memlog== res)
+    def _generate_testcase_callback(self, state, name, message):
```

Comment:
is there any particular reason we override `_generate_testcase_callback` vs modifying `EVMWorld.generate_workspace_files()`?

Reply:
Yes I wanted to avoid calling
testcase_id = self._output.save_testcase(state, name, message)

Which in turns call stuff in workspace module that's too linux biased.
I think maybe workpsace should not do the formatting of any output.

Reply:
`output.save_testcase` generates:
- summary
- trace
- constraints
- input symbols
- state.pkl

all of these are fairly generic on not linux based, no? except summary, which currently has a hack for an evm summary. `save_testcase` then calls into the platform for the platform specific files.

Reply:
constraints, input symbols, state.pkl yes.

'trace' should be the result of asking a plugin to generate the report for it.
for p in plugins:
   dump(p.generate_workspace_files())

Also that trace plugin wont work for EVM as a trace in evm needs a pair. ADDRESS, OFFSET (not state.platform.current.pc as it is now)